docs/src/sources-files.md

```diff
@@ -11,47 +11,59 @@ martin  /path/to/mbtiles/file.mbtiles  /path/to/directory   https://example.org/
+For mbtiles or local pmtiles files, we support auto discovering at startup.
+This means that the following will discover all mbtiles and pmtiles files in the directory:
 
-By default, Martin will use default profile's credentials unless these [AWS environment variables](https://docs.aws.amazon.com/sdkref/latest/guide/creds-config-files.html) are set:
-
-- `AWS_ACCESS_KEY_ID`
-- `AWS_SECRET_ACCESS_KEY`
-- `AWS_SESSION_TOKEN`
-- `AWS_PROFILE` - to specify profile instead of access key variables
-- `AWS_REGION` - if set, must match the region of the bucket in the S3 URI
+```bash
+martin  /path/to/directory
+```
 
-#### Anonymous credentials
+> [!WARNING]
+> For remote PMTiles, we don't currently support auto-discovery.
+> If you want to implement this feature, please see <https://github.com/maplibre/martin/issues/2180>
+>
+> We also don't currently support refreshing the catalog at runtime.
+> If you want to implement this feature, please see <https://github.com/maplibre/martin/issues/288> instead.
 
-By default, martin does require credentials for S3 buckets.
-To send requests anonymously for publicly available buckets, set the environment variable `AWS_SKIP_CREDENTIALS=1` or configuration key `skip_credentials: true` respectively.
+### Serving PMTiles from Object Storage
 
-Note: you still need to set `AWS_REGION` to the correct region.
+Next to local files and remote HTTP sources, we support serving PMTiles from object storage.
+All major cloud providers, including AWS S3, Google Cloud Storage, and Azure Blob Storage are supported.
+The providers differ in the options they support.
 
-Example configuration:
+To serve PMTiles from a provider, you need to provide the bucket name and the prefix of the object key.
+For example:
 
-```yaml
-pmtiles:
-  skip_credentials: false
-  sources:
-    tiles: s3://bucket/path/to/tiles.pmtiles
+```bash
+martin  s3://my-bucket/tiles.pmtiles
 ```
 
-#### Url styles
+The available url schemes are:
 
-We also support forcing path style URLs for S3 buckets via the environment variable `AWS_S3_FORCE_PATH_STYLE=1` or configuration key `force_path_style: true`.
-This allows you to use this functionality for [`MinIO`](https://min.io/) or similar s3-compatible instances which use path style URLs.
-A path style URL is a URL that uses the bucket name as part of the path (`example.org/some_bucket`) instead of the hostname (`some_bucket.example.org`).
+- `file:///path/to/my/file` -> local file system
+- `path/to/my/file` -> local file system
+- `https://mydomain/path` -> Http(s) Source
+- `s3://bucket/path` -> Amazon S3 (also supports `s3a`)
+- `gs://bucket/path` -> Google Cloud Storage
+- `az://account/container/path` -> Microsoft Azure (also supports `adl`, `azure`, `abfs`, `abfss`)
 
-Example configuration:
+If you want more control over your request, you can configure additional options here as such:
 
 ```yaml
 pmtiles:
-  force_path_style: true
+  allow_http: true
```

Comment:
this needs to be documented -- even I have no clue about this parameter, e.g. does it mean remote files are allowed/disable (e.g. local only vs local+remote?), or specifically http, or both http and https?  Please list every possible allowed params, and moreover, i feel this should all go to some dedicated pmtiles doc page as other file sources are likely unrelated to all these configurations